### PR TITLE
Add possibility to leave colorfield blank

### PR DESCRIPTION
**Pull request by SimonBarendse**
Hi,
First of all, thanks for a great extension! I enjoy it very much.
As I've noticed some others required as well (#28), I am looking for an option to leave the colorfield blank.

This is a first draft of a possible solution. If you can spare the time I would love to get some feedback.

For now the extra checkbox indicating whether a value has been set is not updated. I would like to hide this and set it to false when the color-input is clicked.

### CODE REVIEW

File: colorful/widgets.py [hunk 1]
```diff
@@ -2,10 +2,56 @@
...
+
+        if blank:
+            return None
+        else:
+            return color
+
+    @staticmethod
+    def render_script(color_id, checkbox_id):
+        return '''
+                <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.1.1/jquery.min.js"></script>
```
Comment (yakky) [inline colorful/widgets.py]: I don't think jquery is needed here. This can be rewritten as plain js.
Injecting jquery dependency here is would make very hard to maintain and a pain to use it

Comment (SimonBarendse) [reply colorful/widgets.py]: Hi @yakky,
Agree that jquery dependency is not necessary. I've updated the PR with plain js.

Cheers

Comment (yakky) [reply colorful/widgets.py]: Nice!

**Comment by yakky** [comment]
@charettes how can I help to get this merged?

**Comment by charettes** [comment]
I'll deal with the build issues tonight but tests should be added for the new code.

**Comment by charettes** [comment]
Alright CI issues are fixed. Rebasing on top of the latest `master` branch or merging it in should get you started toward adding full coverage for your changes.

**Comment by yakky** [comment]
@SimonBarendse do you need any help on this?

**Comment by SimonBarendse** [comment]
@yakky I currently do not have a lot of spare time. If you would be willing to help out with writing tests, that would be great. More suggestions for improvements are welcome too :)

I've rebased on master, CI seems to work now.

**Comment by yakky** [comment]
@SimonBarendse will do

**Comment by SimonBarendse** [comment]
@yakky Had some free time after all. I wrote some tests, but one seems to fail.

Looks like checking the checkbox propagates to the color input.

**Comment by SimonBarendse** [comment]
@yakky Had some free time after all :)

**Comment by charettes** [comment]
Hey @SimonBarendse thanks for the tests. I had a question, if the checkbox is hidden how is a user supposed to mark the field blank once a color is selected?

**Comment by SimonBarendse** [comment]
Hi @charettes good question. That is currently not possible. When the user selects a color the checkbox is unchecked and the color is used (instead of blank).

With the checkbox hidden, there is currently no possibility to set the field to None again once a color is selected.

To provide that functionality, maybe we should make it optional to make the checkbox visible to the user?

**Comment by charettes** [comment]
I feel like having an always visible checkbox with an `empty` label that would disable the color input when checked and renable it when unchecked would make the most sense here.

**Comment by SimonBarendse** [comment]
Sounds good! Will look into that later

**Comment by guettli** [comment]
What are the next steps? It would be nice to have this feature.

**Comment by charettes** [comment]
@guettli https://github.com/charettes/django-colorful/pull/39#issuecomment-380223695

**Comment by guettli** [comment]
I won't work on this issue. If nobody does, then it might makes sense to close it?

**Comment by SimonBarendse** [comment]
Thanks all for providing feedback on this PR! I'm sorry for leaving this PR without any response from my side for so long.

I no longer have a use case for this and have no plans on continuing this PR. If anyone else does (still) have a use for this functionality, please feel free to re-open and take over. For now, I'll close this PR.
